src/base/net/stratum/SelfSelectClient.cpp

```diff
@@ -235,6 +239,37 @@ void xmrig::SelfSelectClient::submitBlockTemplate(rapidjson::Value &result)
     });
 }
 
+int64_t xmrig::SelfSelectClient::submit(const JobResult& result)
+{
+    if (m_submit_to_origin) {
+        this->submitOriginDaemon(result);
```

Comment:
As far as I can see, this will spam the origin daemon with every self-select pool share without checking the difficulty. Also, `this->` part is probably redundant.

Reply:
Thanks for your comments. Agreed. Will be posting a PR to the branch to correct this.

Reply:
> As far as I can see, this will spam the origin daemon with every self-select pool share without checking the difficulty. 

Thanks yes, I updated the information (in next commit) sent back to the origin daemon to include achieved difficulty as well.

> Also, `this->` part is probably redundant.

Thanks, fixed in next commit.

Reply:
@SChernykh  This has been addressed.